
# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1233768235 18000
# Node ID 11112baa8592f5be016ea748ff4601c7b3c1e3b9
# Parent eb9c1c78e91950557d4c9fd525b5dcc7a56cb23c
auth_debug_passwords=yes: Log password for PAM lookups.

--- a/src/auth/passdb-pam.c	Tue Feb 03 18:44:31 2009 -0500
+++ b/src/auth/passdb-pam.c	Wed Feb 04 12:23:55 2009 -0500
@@ -175,11 +175,15 @@ static int try_pam_auth(struct auth_requ
 			/* log this as error, since it probably is */
 			str = t_strdup_printf("%s (%s missing?)", str, path);
 			auth_request_log_error(request, "pam", "%s", str);
+		} else if (status == PAM_AUTH_ERR) {
+			str = t_strconcat(str, " (password mismatch?)", NULL);
+			if (request->auth->verbose_debug_passwords) {
+				str = t_strconcat(str, " (given password: ",
+						  request->mech_password,
+						  ")", NULL);
+			}
+			auth_request_log_info(request, "pam", "%s", str);
 		} else {
-			if (status == PAM_AUTH_ERR) {
-				str = t_strconcat(str, " (password mismatch?)",
-						  NULL);
-			}
 			auth_request_log_info(request, "pam", "%s", str);
 		}
 		return status;

